packages/table/src/Row.tsx

```diff
@@ -0,0 +1,323 @@
+      const chevronButton = (
+        <IconButton
+          onClick={() => setIsExpanded(curr => !curr)}
+          aria-label="chevron"
+          className={iconButtonMargin}
+        >
+          <Icon aria-label="chevron" color={uiColors.gray.dark2} />
+        </IconButton>
+      );
+      const renderedChildren: Array<React.ReactElement> = [];
+      let hasSeenRow = false;
+
+      React.Children.forEach(children, (child, index) => {
+        if (isComponentType(child, 'Cell')) {
+          if (!child.props.children) {
+            return null;
+          }
+
+          renderedChildren.push(
+            React.cloneElement(child, {
+              children: (
+                <span className={truncation}>{child.props.children}</span>
+              ),
+              key: `${indexRef.current}-${index}`,
+              disabled: child.props.disabled || disabled,
+            }),
+          );
+        } else {
+          if (isComponentType(child, 'Row')) {
+            hasSeenRow = true;
+          }
+        }
+      });
+
+      if (hasSeenRow) {
+        renderedChildren[0] = React.cloneElement(renderedChildren[0], {
+          children: (
+            <>
+              {chevronButton}
+              <span className={truncation}>
+                {renderedChildren[0].props.children}
+              </span>
+            </>
+          ),
+          className: cx(displayFlex, className),
+          key: `${indexRef.current}-${renderedChildren[0].props.children}`,
+        });
+      }
+
+      return renderedChildren;
+    }, [children, disabled, className, isExpanded, setIsExpanded]);
+
+    // Iterating over children twice because generated memoized values have different dependants
+    const nestedRows = React.useMemo(() => {
+      let hasSeenFirstCell = false;
+      const nestedRows: Array<React.ReactElement> = [];
+
+      React.Children.forEach(children, (child, index) => {
+        if (isComponentType(child, 'Cell') && !hasSeenFirstCell) {
+          hasSeenFirstCell = true;
+        }
+
+        if (child != null && isComponentType(child, 'Row')) {
+          nestedRows.push(
+            React.cloneElement(child, {
+              ref: nodeRef,
+              isAnyAncestorCollapsed: isAnyAncestorCollapsedProp || !isExpanded,
+              indentLevel: indentLevel + 1,
+              key: `${indexRef.current}-${indentLevel}-${index}`,
+            }),
+          );
+        }
+      });
+
+      return nestedRows;
+    }, [isAnyAncestorCollapsedProp, isExpanded, indentLevel, children]);
+
+    // Depending on network speed, will noticeably render columns with incorrect
+    // alignment, would rather wait for proper information before rendering
+    if (!columnInfo) {
+      return null;
+    }
+
+    const shouldAltRowColor = data && data.length >= 10 && !hasNestedRows;
+
+    const alignmentStyles = Object.entries(
+      columnInfo,
+    ).map(([key, { dataType }]) => styleColumn(key, dataType!));
```

Comment:
We should fix the argument syntax of `styleColumn` as `dataType` can absolutely be undefined, we're not requiring them set that on columns are we?

Reply:
Fixed!